BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m

```diff
@@ -400,12 +410,17 @@ - (NSURLSessionUploadTask *)backgroundUploadTaskWithRequest:(NSURLRequest *)requ
 
 - (BOXURLSessionTaskCachedInfo *)sessionTaskCompletedCachedInfoGivenUserId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)error
 {
-    return [self.cacheClient completedCachedInfoForUserId:userId associateId:associateId error:error];
+    return self.supportBackgroundSessionTasks == NO ? nil : [self.cacheClient completedCachedInfoForUserId:userId associateId:associateId error:error];
+}
+
+- (BOOL)supportBackgroundSessionTasks
+{
+    return self.backgroundSession != nil && self.cacheClient != nil;
 }
 
 - (BOOL)cleanUpSessionTaskInfoGivenUserId:(NSString *)userId associateId:(NSString *)associateId error:(NSError **)outError
 {
-    if (userId == nil || associateId == nil) {
+    if (self.supportBackgroundSessionTasks == NO || userId == nil || associateId == nil) {
```

Comment:
this only applies for background session task, will update the title accordingly